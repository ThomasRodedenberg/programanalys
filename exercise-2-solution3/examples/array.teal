fun main() = {
    var a : nonnull array[nonnull int] := [1, 2, 3, 4]; // some stuff,  N 2 4 +

    var x := a[1]; // NI 4 13 +, N 4 4 = 
    var y := a[null]; // NI 5 13 0, N 5 5 =
}