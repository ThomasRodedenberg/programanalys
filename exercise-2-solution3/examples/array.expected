maybe: NI 2 46 +
maybe: NI 2 49 +
maybe: NI 2 52 +
maybe: NI 2 4 +
N 2 4 +
NI 4 13 +
NI 5 13 0
